### PR TITLE
feat(event-bridge): adding in event bridge utility package

### PR DESCRIPTION
**Pull request by bassrock**
# Goal

Create an event bridge utility package for working with the Event Bridge types on the Pocket event bridge.

### CODE REVIEW

File: lambdas/transactional-emails/src/handlers/forgotPassword.ts [hunk 1]
```diff
@@ -50,14 +9,14 @@ export function validateEventPayload(
  * @throws Error if response is not ok
  */
 export async function forgotPasswordHandler(record: SQSRecord) {
-  const message = JSON.parse(JSON.parse(record.body).Message)['detail'];
-
-  const event = validateEventPayload(message);
-
+  const event = sqsEventBridgeEvent(record);
```
Comment (bassrock) [inline lambdas/transactional-emails/src/handlers/forgotPassword.ts:12]: this is the magic area of the code

File: packages/event-bridge/src/events/forgotPasswordRequest.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { EventBridgeBase } from './eventBridgeBase';
+import { Event } from './names';
+
+export interface ForgotPasswordRequest extends EventBridgeBase {
```
Comment (bassrock) [inline packages/event-bridge/src/events/forgotPasswordRequest.ts:4]: Example of what a type will look like in event bridge

File: packages/event-bridge/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  if (!json || !(json['detail-type'] in Object.keys(Event))) {
+    throw new Error(`Unsupported type: ${json['detail-type']}`);
+  }
+  return json as PocketEventTypeMap[T];
+};
+
+export const sqsEventBridgeEvent = <T extends keyof PocketEventTypeMap>(
+  record: SQSRecord,
+): PocketEventTypeMap[T] | null => {
+  const message = JSON.parse(JSON.parse(record.body).Message);
```
Comment (bassrock) [inline packages/event-bridge/src/utils.ts:17]: Function to take an SQS message and turn it into a useable object

File: lambdas/transactional-emails/src/handlers/forgotPassword.ts [hunk 4]
```diff
@@ -1,47 +1,6 @@
...
-import { FogotPasswordResetEvent } from '../schemas/forgotPasswordResetEvent';
-
-/**
- * Validates that email is not null in the forgot password event payload and
- * that we have valid reset information
- * @param message forgot password reset event forwarded from event bridge
- * @returns FogotPasswordResetEvent
- * @throws Error if email is missing in the event payload
- */
-export function validateEventPayload(
```
Comment (bassrock) [inline lambdas/transactional-emails/src/handlers/forgotPassword.ts:12]: buh bye.

File: packages/event-bridge/src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  // TODO: Event validation of fields
```
Comment (bassrock) [inline packages/event-bridge/src/utils.ts:18]: Open to validation ideas

**Review by kschelonka — APPROVED** [review]
hero, I've been wanting this for forever

**Pull request merged**